liiatools_pipeline/sensors/job_success_sensor.py

```diff
@@ -19,14 +19,14 @@
 from liiatools_pipeline.ops.common_config import CleanConfig
 
 
-def find_previous_matching_ssda903_run(run_records):
+def find_previous_matching_dataset_run(run_records, dataset):
     previous_run_id = [
         run.dagster_run.run_id
         for run in run_records
-        if "ssda903" == run.dagster_run.tags["dataset"]
+        if dataset == run.dagster_run.tags["dataset"]
```

Comment:
Could we get a try/except block around this in case the key doesn't exist in the run tag data? Older jobs won't have this (especially in staging and production) and it'd be good to protect against this being an issue.